modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h_fp.i

```diff
@@ -0,0 +1,278 @@
+  []
+  [u_pressure]
+    type = LinearFVMomentumPressure
+    variable = vel_x
+    pressure = pressure
+    momentum_component = 'x'
+  []
+
+  [p_diffusion]
+    type = LinearFVAnisotropicDiffusion
+    variable = pressure
+    diffusion_tensor = Ainv
+    use_nonorthogonal_correction = false
+  []
+  [HbyA_divergence]
+    type = LinearFVDivergence
+    variable = pressure
+    face_flux = HbyA
+    force_boundary_execution = true
+  []
+
+  [temp_advection]
+    type = LinearFVEnergyAdvection
+    variable = h
+  []
+  [source]
+    type = LinearFVSource
+    variable = h
+    source_density = source_func
+  []
+[]
+
+[LinearFVBCs]
+  [inlet_u]
+    type = LinearFVAdvectionDiffusionFunctorDirichletBC
+    boundary = 'left'
+    variable = vel_x
+    functor = ${bulk_u}
+  []
+  [inlet_h]
+    type = LinearFVAdvectionDiffusionFunctorDirichletBC
+    variable = h
+    boundary = 'left'
+    functor = 'h_from_p_T'
+  []
+  [inlet_T]
+    type = LinearFVAdvectionDiffusionFunctorDirichletBC
+    variable = T
+    boundary = 'left'
+    functor = ${T_in}
+  []
+
+  [outlet_p]
+    type = LinearFVAdvectionDiffusionFunctorDirichletBC
+    boundary = 'right'
+    variable = pressure
+    functor = ${p_ref}
+  []
+  [outlet_h]
+    type = LinearFVAdvectionDiffusionOutflowBC
+    variable = h
+    use_two_term_expansion = false
+    boundary = 'right'
+  []
+  [outlet_u]
+    type = LinearFVAdvectionDiffusionOutflowBC
+    variable = vel_x
+    use_two_term_expansion = false
+    boundary = 'right'
+  []
+[]
+
+[FluidProperties]
+  [lead]
+    type = LeadFluidProperties
+  []
+[]
+
+[FunctorMaterials]
+  [enthalpy_material]
+    type = LinearFVEnthalpyFunctorMaterial
+    pressure = ${p_ref}
+    T_fluid = T
+    h = h
+    fp = lead
+  []
+  [fluid_props_to_mat_props]
+    type = GeneralFunctorFluidProps
+    fp = lead
+    pressure = ${p_ref}
+    T_fluid = 'T'
+    speed = 1
+    porosity = 1
+    characteristic_length = 1
+  []
+  [source_func]
+    type = ADParsedFunctorMaterial
+    property_name = source_func
+    functor_names = 'rho'
+    expression = ${q_source}
+  []
+[]
+
+[AuxKernels]
+  [rho_out]
+    type = FunctorAux
+    functor = 'rho'
+    variable = 'rho_var'
+    execute_on = 'NONLINEAR'
+  []
+  [cp_out]
+    type = FunctorAux
+    functor = 'cp'
+    variable = 'cp_var'
+    execute_on = 'NONLINEAR'
+  []
+  [mu_out]
+    type = FunctorAux
+    functor = 'mu'
+    variable = 'mu_var'
+    execute_on = 'NONLINEAR'
+  []
+  [k_out]
+    type = FunctorAux
+    functor = 'k'
+    variable = 'k_var'
+    execute_on = 'NONLINEAR'
+  []
+  [T_from_h_functor_aux]
+    type = FunctorAux
+    functor = 'T_from_p_h'
+    variable = 'T'
+    execute_on = 'NONLINEAR'
+  []
+  [h_from_T_functor_aux]
+    type = FunctorAux
+    functor = 'h_from_p_T'
+    variable = 'h_aux'
+    execute_on = 'NONLINEAR'
+  []
+[]
+
+[Postprocessors]
+  [T_out_sim]
+    type = ElementalVariableValue
+    variable = T
+    elementid = ${fparse nx-1}
```

Comment:
this might not work if you have a distributed mesh (the numbering can be local). Trying a point value postprocessor might be better in that sense